components/date-picker/picker.component.ts

```diff
@@ -288,8 +292,17 @@ export class NzPickerComponent implements OnInit, AfterViewInit, OnChanges, OnDe
 
   updateInputWidthAndArrowLeft(): void {
     this.inputWidth = this.rangePickerInputs?.first?.nativeElement.offsetWidth || 0;
-    this.datePickerService.arrowLeft =
-      this.datePickerService.activeInput === 'left' ? 0 : this.inputWidth + this.separatorElement?.nativeElement.offsetWidth || 0;
+
+    if (this.dir === 'rtl') {
+      this.datePickerService.arrowRight =
+        this.datePickerService.activeInput === 'right' ? this.inputWidth + this.separatorElement?.nativeElement.offsetWidth || 0 : 0;
+      this.datePickerService.arrowLeft = 'auto';
+    } else {
+      this.datePickerService.arrowLeft =
+        this.datePickerService.activeInput === 'left' ? 0 : this.inputWidth + this.separatorElement?.nativeElement.offsetWidth || 0;
+      this.datePickerService.arrowRight = 'auto';
+    }
+
```

Comment:
```suggestion

    const baseStyle = { position: 'absolute', width: `${this.inputWidth}px` };
    this.datePickerService.arrowLeft = this.datePickerService.activeInput === 'left' ? 0 : this.inputWidth + this.separatorElement?.nativeElement.offsetWidth || 0;

    if (this.dir === 'rtl') {
      this.positionStyle = { ...baseStyle, right: `${this.datePickerService.arrowLeft}px` };
    } else {
      this.positionStyle = { ...baseStyle, left: `${this.datePickerService.arrowLeft}px` };
    }

```

also need to define `positionStyle`.